staging/src/k8s.io/apimachinery/pkg/util/wait/wait_test.go

```diff
@@ -455,18 +455,46 @@ func TestWaitFor(t *testing.T) {
 	}
 }
 
+func TestWaitForWithClosedChannel(t *testing.T) {
+	stopCh := make(chan struct{})
+	close(stopCh)
+	start := time.Now()
+	err := WaitFor(poller(ForeverTestTimeout, ForeverTestTimeout), func() (bool, error) {
+		return false, nil
+	}, stopCh)
+	duration := time.Now().Sub(start)
+	// The WaitFor should returns immediately. So the duration is closed to 0s.
+	// This condition ensures that if the WaitFor returns error caused by poller rather
+	// than stopCh, it will trigger an error.
+	if duration >= ForeverTestTimeout/2 {
+		t.Errorf("expected short timeout duration")
+	}
+	if err != ErrWaitTimeout {
+		t.Errorf("expected ErrWaitTimeout from WaitFunc")
+	}
+}
+
 func TestWaitForWithDelay(t *testing.T) {
```

Comment:
IIUC, you are trying to verify that if the WaitFor's stopCh is closed, then the done channel pass to the waitFunc is also closed. If so, can you rename the function and variable names to be more clear?